mypy/semanal.py

```diff
@@ -1643,26 +1623,59 @@ def alias_fallback(self, tp: Type) -> Instance:
         fb_info.mro = [fb_info, self.object_type().type]
         return Instance(fb_info, [])
 
+    def analyze_alias(self, rvalue: Expression,
+                      allow_unnormalized: bool) -> Tuple[Optional[Type], List[str]]:
+        res = analyze_type_alias(rvalue,
+                                 self.lookup_qualified,
+                                 self.lookup_fully_qualified,
+                                 self.tvar_scope,
+                                 self.fail, self.plugin, allow_unnormalized=True)
+        if res:
+            alias_tvars = [name for (name, _) in
+                           res.accept(TypeVariableQuery(self.lookup_qualified, self.tvar_scope))]
+        else:
+            alias_tvars = []
+        return res, alias_tvars
+
     def check_and_set_up_type_alias(self, s: AssignmentStmt) -> None:
         """Check if assignment creates a type alias and set it up as needed."""
-        # For now, type aliases only work at the top level of a module.
+        # Type aliases are created only at module scope, at class and function scopes
+        # assignments create class valued members and local variables with type object types.
```

Comment:
This caused an error in an internal Dropbox codebase. We have code that looks like this:
```py
from typing import Union

class A:
    B = Union[int, str]

    def f(self) -> B: # this PR makes this an invalid type
        pass
```

Maybe type aliases in the class body should still be supported? Alternatively, maybe anything where the rvalue is a valid type that is an `IndexExpr` should still be considered as a type alias, i.e. only restrict simple cases like `X = Y`.